### PR TITLE
Feature: final day assignment screen (kids-955)

### PR DESCRIPTION
**Pull request by Daniela510**

### CODE REVIEW

File: lib/features/children/generosity_challenge/pages/generosity_challenge_day_details.dart [hunk 1]
```diff
@@ -70,10 +70,11 @@ class GenerosityChallengeDayDetails extends StatelessWidget {
                     ),
                   ),
                 ),
-              if (!isDualCard ||
-                  state.status ==
-                      GenerosityChallengeStatus.dailyAssigmentConfirm ||
-                  day.isCompleted)
+              if (challenge.state.islastDay &
+                  (!isDualCard ||
```
Comment (TammiLion) [inline lib/features/children/generosity_challenge/pages/generosity_challenge_day_details.dart:74]: For context, what does "isDualCard" mean, when does this occur in the app/challenge?

Comment (Daniela510) [reply lib/features/children/generosity_challenge/pages/generosity_challenge_day_details.dart:74]: It occurs on day 2 and day 7, it is when the assignment screen does not have a complete button. They see an assignment the follow a flow (select values/ give to org) then they come back to the assignment screen, see a dynamic descriptio and can click on the complete button. But i will also leave a comment in the code, try to make that logic clearer

Comment (TammiLion) [reply lib/features/children/generosity_challenge/pages/generosity_challenge_day_details.dart:74]: Clear, thanks!

**Review by TammiLion — APPROVED** [review]

File: lib/features/children/generosity_challenge/pages/generosity_challenge_day_details.dart [hunk 2]
```diff
@@ -23,7 +23,20 @@ class GenerosityChallengeDayDetails extends StatelessWidget {
...
+    final isDailyAssignmentConfirm = challenge.state.status ==
+        GenerosityChallengeStatus.dailyAssigmentConfirm;
+    final isDayCompleted = day.isCompleted;
+    // We do NOT show the 'Complete' button if:
+    // - the challenge is on the last day
+    // We show the 'Complete' button if:
+    // - it is a single card task (day 3- 6)
+    // - the day is already completed (disabled state)
+    // - the status is dailyAssigmentConfirm (happens for dual cards day 2/7)
+    final shouldShowCompleteButton = !isLastDay &
```
Comment (TammiLion) [inline lib/features/children/generosity_challenge/pages/generosity_challenge_day_details.dart:37]: Perfection 👌

**Pull request merged**